### PR TITLE
feat:(login) add login using email and password and return access_token

### PR DESCRIPTION
**Pull request by mariodenny**

### CODE REVIEW

File: db/migrations/first_init_db_table.sql [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
     id BIGINT AUTO_INCREMENT PRIMARY KEY NOT NULL,
-    fullname VARCHAR(255),
-    sort_bio VARCHAR(255),
-    gender ENUM('male','female','others'),
+    fullname VARCHAR(255) NULL,
+    sort_bio VARCHAR(255) NULL,
+    gender ENUM('male','female','others') DEFAULT 'others',
     picture_url VARCHAR(255) DEFAULT NULL,
-    username VARCHAR(255),
+    username VARCHAR(255) NULL,
```
Comment (rezaig) [inline db/migrations/first_init_db_table.sql]: please edit username to not nullable

File: src/model/account.go [hunk 2]
```diff
@@ -2,12 +2,14 @@ package model
 
 import (
 	"context"
-	"mime/multipart"
 	"time"
 )
 
 type AccountRepository interface {
-	RegisterNewAccountToDatabase(ctx context.Context, data Account) (*Account, error)
+	StoreAccount(ctx context.Context, data Account) (*Account, error)
```
Comment (rezaig) [inline src/model/account.go]:
```suggestion
	Store(ctx context.Context, data Account) (*Account, error)
```
Don't repeat the context "Account", it is already represented in "AccountRepository" interface

File: src/model/account.go [hunk 3]
```diff
@@ -40,22 +42,18 @@ type Account struct {
 }
 
 type AccountUsecase interface {
-	CreateNewAccountData(ctx context.Context, data Register, fileHeader *multipart.FileHeader) (token string, err error)
+	CreateAccount(ctx context.Context, data Register) (token string, err error)
```
Comment (rezaig) [inline src/model/account.go]:
```suggestion
	Create(ctx context.Context, data Register) (token string, err error)
```

File: src/repository/account_repository.go [hunk 4]
```diff
@@ -53,4 +54,28 @@ func (r *accountRepository) RegisterNewAccountToDatabase(ctx context.Context, da
...
+	err = row.Scan(
+		&data.ID,
+		&data.Email,
+		&data.Password,
+	)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			logrus.WithField("email", email).Error("email not found")
+			return nil, fmt.Errorf("email %s not found", email)
+		}
```
Comment (rezaig) [inline src/repository/account_repository.go]: better don't return error if data is not found or empty, just return nil to the front. then let usecase layer consider the flow

File: src/usecase/account_usecase.go [hunk 5]
```diff
@@ -63,7 +51,28 @@ func (account *accountUsecase) CreateNewAccountData(ctx context.Context, data mo
...
+	logger := logrus.WithFields(logrus.Fields{
+		"data": data.Email,
+	})
+
+	user, err := u.accountRepository.FindByEmail(ctx, data.Email)
+
+	if err != nil {
+		logger.Error(err)
+		return
+	}
```
Comment (rezaig) [inline src/usecase/account_usecase.go]:
```suggestion
	user, err := u.accountRepository.FindByEmail(ctx, data.Email)
	if err != nil {
		logger.Error(err)
		return
	}
```
don't add a newline

File: src/usecase/account_usecase.go [hunk 6]
```diff
@@ -63,7 +51,28 @@ func (account *accountUsecase) CreateNewAccountData(ctx context.Context, data mo
...
+	if !helper.CheckPasswordHash(data.Password, user.Password) {
+		err = errors.New("incorrect password")
+		return
+	}
+
+	token, err = helper.GenerateToken(user.ID)
+
+	if err != nil {
+		logger.Error(err)
+	}
```
Comment (rezaig) [inline src/usecase/account_usecase.go:71]:
```suggestion
	token, err = helper.GenerateToken(user.ID)
	if err != nil {
		logger.Error(err)
	}
```

File: src/usecase/account_usecase.go [hunk 7]
```diff
@@ -18,33 +17,22 @@ func NewAccountUsecase(accountRepository model.AccountRepository) model.AccountU
...
-		logger.Error(err)
-		return "", fmt.Errorf("failed to save uploaded file: %w", err)
-	}
-	data.PictureUrl = imageUploaded
 	passwordHashed, err := helper.HashRequestPassword(data.Password)
 
 	if err != nil {
 		logger.Error(err)
 		return
 	}
```
Comment (rezaig) [inline src/usecase/account_usecase.go]:
```suggestion
	passwordHashed, err := helper.HashRequestPassword(data.Password)
	if err != nil {
		logger.Error(err)
		return
	}
```

File: src/handler/http/account_handler.go [hunk 8]
```diff
@@ -16,9 +16,10 @@ func NewAccountHandler(accountUsecase model.AccountUsecase) *AccountHandler {
 }
 
 func (handler *AccountHandler) RegisterRoute(e *echo.Echo) {
-	groupping := e.Group("accounts")
+	groupping := e.Group("auth")
 
 	groupping.POST("/register", handler.register)
+	groupping.POST("/login", handler.login)
```
Comment (rezaig) [inline src/handler/http/account_handler.go]:
```suggestion
        groupping := e.Group("/v1/auth")

	e.POST("/register", handler.register)
	e.POST("/login", handler.login)
```

**Review by rezaig — CHANGES_REQUESTED** [review]

File: db/migrations/first_init_db_table.sql (hunk 1 above)
Comment (mariodenny) [reply db/migrations/first_init_db_table.sql]: Done  mas

File: src/model/account.go (hunk 2 above)
Comment (mariodenny) [reply src/model/account.go]: Done

**Pull request merged**